src/main/java/com/zegoggles/smssync/preferences/Preferences.java

```diff
@@ -161,8 +163,17 @@ public int getRegularTimeoutSecs() {
         return getStringAsInt(REGULAR_TIMEOUT_SECONDS, Defaults.REGULAR_TIMEOUT_SECONDS);
     }
 
-    public boolean getMarkAsRead() {
-        return preferences.getBoolean(MARK_AS_READ.key, Defaults.MARK_AS_READ);
+    public void migrateMarkAsRead() {
+        if (preferences.contains(MARK_AS_READ.key)) {
+            if (!preferences.getBoolean(MARK_AS_READ.key, true)) {
+                preferences.edit().putString(MARK_AS_READ_TYPES.key, "unread").commit();
```

Comment:
`getMarkAsReadType()` has a default value, so I don't think this is really needed. But I can change it, if you want.